pkgs/apps/dirac/default.nix

```diff
@@ -0,0 +1,154 @@
+    python3
+    makeWrapper
+  ];
+
+  # For now, DIRAC is built without the PCMSolver optional dependency.
+  buildInputs = [
+    blas
+    lapack
+    boost
+    exatensor
+  ];
+
+  propagatedBuildInputs = [
+    mpi
+    which
+  ];
+
+  src = fetchurl {
+    url = "https://zenodo.org/record/4836496/files/DIRAC-${version}-Source.tar.gz";
+    sha256 = "0vqmin24xqkcp3rvml0dimc4kjmb3m37c3l6bkjy4y6d3xyb41yg";
+  };
+
+  postPatch = ''
+    # cd $src
+    patchShebangs .
+
+    cat > cmake/custom/exatensor.cmake << EOF
+    # Patched CMAKE file that does not download any exatensor.
+
+    # ExternalProject_Add seems required as later in the build process,
+    # DIRAC includes exatensor via the CMAKE dependency system. It's not 100%
+    # clear to me, so I'll just have that as a dummy here.
+    ExternalProject_Add(exatensor
+        PREFIX exatensor-tmp
+        INSTALL_DIR ${exatensor}
+        STAMP_DIR exatensor-tmp
+        SOURCE_DIR exatensor-tmp/src
+        CONFIGURE_COMMAND true
+        BUILD_COMMAND true
+        INSTALL_COMMAND true
+    )
+
+    set(EXTERNAL_LIBS ''${EXTERNAL_LIBS} ${exatensor}/lib/libtalsh.so)
+    set(EXTERNAL_LIBS ''${EXTERNAL_LIBS} ${exatensor}/lib/libexatensor.so)
+
+    include_directories(${exatensor}/include)
+
+    # DIRAC includes some exatensor-related tests directly here.
+    # I'll just leave them in.
+    dirac_test(exacorr_talsh_debug "cc;talsh;short" "")
+    dirac_test(exacorr_talsh_standalone "cc;talsh;short" "")
+    dirac_test(exacorr_talsh_lambda "cc;talsh" "")
+    dirac_test(exacorr_talsh_fock "cc;talsh" "")
+    dirac_test(exacorr_talsh_open "cc;talsh;short" "")
+    dirac_test(exacorr_talsh_cc2 "cc;talsh" "")
+    dirac_test(exacorr_talsh_spinfree "cc;talsh" "")
+    dirac_test(exacorr_talsh_respect "cc;talsh;respect" "")
+    dirac_test(exacorr_talsh_memory "cc;talsh;short" "")
+    if (ENABLE_MPI AND NOT EXATENSOR_TALSH)
+        dirac_test(exacorr_exatensor_debug "cc;exatensor" "")
+        dirac_test(exacorr_exatensor_fock "cc;exatensor" "")
+        dirac_test(exacorr_exatensor_open "cc;exatensor" "")
+        dirac_test(exacorr_exatensor_lambda "cc;exatensor" "")
+        dirac_test(exacorr_exatensor_cc2 "cc;exatensor" "")
+        dirac_test(exacorr_exatensor_memory "cc;exatensor;short" "")
+    endif()
+
+    EOF
+  '';
+
+  FC = "mpif90";
+  CC = "mpicc";
+  CXX = "mpicxx";
+  MATH_ROOT = blas;
+  ENABLE_EXATENSOR = "ON";
+
+  /*
+  Cmake is required to build but adding it to the buildinputs then ignores the
+  setup script. Therefore i call the script here manually but cmake is invoked
+  by setup.
+  */
+
+  configurePhase = ''
+    ./setup --prefix=$out --mpi && cd build
+  '';
+
+  enableParallelBuilding = true;
+  hardeningDisable = [ "format" ];
+  doInstallCheck = true;
+
+  /*
+  Make the MPI stuff available to the DIRAC script. Direct exposure of MPI is not necessary.
+  */
+  postFixup = ''
+    wrapProgram $out/bin/pam-dirac \
+      --prefix PATH : ${mpi}/bin \
```

Comment:
Is `pam-dirac` just a script (not a binary)? If yes, you could use `substituteInPlace` to directly hard code the path instead of using a wrapper.